roundabout-core/src/test/java/at/fhv/itm3/s2/roundabout/integration/DestinationReachedIntegration.java

```diff
@@ -0,0 +1,54 @@
+package at.fhv.itm3.s2.roundabout.integration;
+
+import at.fhv.itm3.s2.roundabout.RoundaboutSimulationModel;
+import at.fhv.itm3.s2.roundabout.api.entity.AbstractSource;
+import at.fhv.itm3.s2.roundabout.api.entity.ICar;
+import at.fhv.itm3.s2.roundabout.api.entity.IRoute;
+import at.fhv.itm3.s2.roundabout.api.entity.Street;
+import at.fhv.itm3.s2.roundabout.mocks.RouteGenerator;
+import at.fhv.itm3.s2.roundabout.mocks.RouteType;
+import desmoj.core.simulator.Experiment;
+import desmoj.core.simulator.TimeInstant;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.mockito.Mockito.when;
+
+public class DestinationReachedIntegration {
+
+    private RoundaboutSimulationModel model;
+    private Experiment exp;
+
+    @Before
+    public void setUp() {
+        model = new RoundaboutSimulationModel(null, "", false, false);
+        exp = new Experiment("RoundaboutSimulationModel Experiment");
+        model.connectToExperiment(exp);
+        exp.setShowProgressBar(false);
+    }
+
+    @Test
+    public void destinationReached() {
+        exp.stop(new TimeInstant(60, TimeUnit.SECONDS));
+
+        RouteGenerator routeGenerator = new RouteGenerator(model);
+        IRoute route = routeGenerator.getRoute(RouteType.TWO_STREETSECTIONS_ONE_CAR);
+        AbstractSource source = route.getSource();
+        Street destinationSection = route.getDestinationSection();
+
+        source.startGeneratingCars();
+
+        exp.start();
+        exp.finish();
+
+        if(!destinationSection.isEmpty()){
+            Assert.assertEquals("car never reached destination.",31, destinationSection.getIdentNumber());
+        } else {
+            Assert.fail();
+        }
```

Comment:
the assert part is quite hard to understand at the first look
remember: there should be no logic in a test, so make sure you test only one thing (either car hast reached destination or not)
so i would e.g. Assert.assertFalse(destinationSection.isEmpty())

then you do not need the fail part because in case the section is empty, the test fails automatically